project-fob/Models/User.cs

```diff
@@ -25,5 +25,31 @@ public User(string id)
             UserId = id;
             Lastcheckin = DateTime.Now;
         }
+        public override bool Equals(object obj)
+        {
+            if (obj is User paramater)
+            {
+                return Equals(paramater);
+            }
+            return false;
+            //return base.Equals(obj);
+        }
+
+        private bool Equals(User parameter)
+        {
+            if (parameter.UserId.Equals(UserId))
+            {
+                    return true;
+            }
+            return false;
+        }
+
+        public override int GetHashCode()
+        {
+            var hashCode = -1403531409;
+            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(UserId);
+            hashCode = hashCode * -1521134295 + Lastcheckin.GetHashCode();
+            return hashCode;
```

Comment:
The only thing being checked in `Equals` is the `UserId`, so `GetHashCode` should also only use UserID. It's important for GetHashCode and Equals to agree on what makes something "the same" bad things can happen. 

Specifically, if two objects return true from `Equals` then they must return the same value as each other from `GetHashCode`, which won't be the case here if only `Lastcheckin` differs between objects.